lib/cache/add-remote-git.js

```diff
@@ -375,14 +383,6 @@ function updateSubmodules (from, resolvedURL, tmpdir, cb) {
         // https://github.com/npm/npm/issues/6400
         addLocal(spec, null, function (er, data) {
           if (data) {
-            if (npm.config.get('save-exact')) {
-              log.verbose('addRemoteGit', 'data._from:', resolvedURL, '(save-exact)')
```

Comment:
I think even with your patch we need this behavior if `--save-exact` was used, as it saves with a commit id.